tests/unittests/CoreGraphics/CGGeometryTests.mm

```diff
@@ -52,4 +52,122 @@
+    ::testing::make_tuple(CGRectMake(0, 0, 900, 800), -502, CGRectMinXEdge, CGRectMake(0, 0, 0, 800), CGRectMake(0, 0, 900, 800)),
+    ::testing::make_tuple(CGRectMake(0, 0, 900, 800),
+                          0.34,
+                          CGRectMaxYEdge,
+                          CGRectMake(0, 799.66, 900, 0.34),
+                          CGRectMake(0, 0, 900, 799.66)),
+    ::testing::make_tuple(CGRectMake(0, 0, 900, 800), 120, CGRectMinYEdge, CGRectMake(0, 0, 900, 120), CGRectMake(0, 120, 900, 680)),
+    ::testing::make_tuple(CGRectMake(10.50, 25.60, 300, 500),
+                          -50,
+                          CGRectMaxXEdge,
+                          CGRectMake(310.50, 25.60, 0, 500),
+                          CGRectMake(10.50, 25.60, 300, 500)),
+    ::testing::make_tuple(CGRectMake(10.50, 25.60, 300, 500),
+                          10,
+                          CGRectMinXEdge,
+                          CGRectMake(10.50, 25.60, 10, 500),
+                          CGRectMake(20.50, 25.60, 290, 500)),
+    ::testing::make_tuple(CGRectMake(10.50, 25.60, 300, 500),
+                          0.34,
+                          CGRectMaxYEdge,
+                          CGRectMake(10.50, 525.26, 300, 0.34),
+                          CGRectMake(10.50, 25.60, 300, 499.66)),
+    ::testing::make_tuple(CGRectMake(10.50, 25.60, 300, 500),
+                          120,
+                          CGRectMinYEdge,
+                          CGRectMake(10.50, 25.60, 300, 120),
+                          CGRectMake(10.50, 145.60, 300, 380)),
+    ::testing::make_tuple(CGRectMake(100, 100, 125, 250),
+                          50.34,
+                          CGRectMaxXEdge,
+                          CGRectMake(174.66, 100, 50.34, 250),
+                          CGRectMake(100, 100, 74.66, 250)),
+    ::testing::make_tuple(CGRectMake(100, 100, 125, 250), 32, CGRectMinXEdge, CGRectMake(100, 100, 32, 250), CGRectMake(132, 100, 93, 250)),
+    ::testing::make_tuple(CGRectMake(100, 100, 125, 250),
+                          20,
+                          CGRectMaxYEdge,
+                          CGRectMake(100, 330, 125, 20),
+                          CGRectMake(100, 100, 125, 230)),
+    ::testing::make_tuple(CGRectMake(100, 100, 125, 250),
+                          -50.24,
+                          CGRectMinYEdge,
+                          CGRectMake(100, 100, 125, 0),
+                          CGRectMake(100, 100, 125, 250)),
+    ::testing::make_tuple(CGRectNull, -50.24, CGRectMaxXEdge, CGRectNull, CGRectNull),
+    ::testing::make_tuple(CGRectNull, 150.240000, CGRectMinXEdge, CGRectNull, CGRectNull),
+    ::testing::make_tuple(CGRectNull, -50.24, CGRectMaxYEdge, CGRectNull, CGRectNull),
+    ::testing::make_tuple(CGRectNull, 150.240000, CGRectMinYEdge, CGRectNull, CGRectNull),
+};
+
+INSTANTIATE_TEST_CASE_P(CGGeometry, CGGeometryDivRect, ::testing::ValuesIn(cCGRectDivideResults));
+
+TEST(CGGeometry, RectDivideInf) {
```

Comment:
it just makes this odd case stand out. also makes life easier + we don't have to compute CGRectInf

---
In reply to: [105253463](https://github.com/Microsoft/WinObjC/pull/2100#discussion_r105253463) [](ancestors = 105253463)